app/backend/src/couchers/models.py

```diff
@@ -341,26 +342,111 @@ class FriendRelationship(Base):
     to_user = relationship("User", backref="friends_to", foreign_keys="FriendRelationship.to_user_id")
 
 
-class SignupToken(Base):
+class ContributeOption(enum.Enum):
+    yes = enum.auto()
+    maybe = enum.auto()
+    no = enum.auto()
+
+
+class ContributorForm(Base):
     """
-    A signup token allows the user to verify their email and continue signing up.
+    Someone filled in the contributor form
     """
 
-    __tablename__ = "signup_tokens"
-    token = Column(String, primary_key=True)
+    __tablename__ = "contributor_forms"
 
-    email = Column(String, nullable=False)
+    id = Column(BigInteger, primary_key=True)
 
-    # timezones should always be UTC
+    user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+
+    ideas = Column(String, nullable=True)
+    features = Column(String, nullable=True)
+    experience = Column(String, nullable=True)
+    contribute = Column(Enum(ContributeOption), nullable=True)
+    contribute_ways = Column(ARRAY(String), nullable=True)
+    expertise = Column(String, nullable=True)
+
+    user = relationship("User", backref="contributor_forms")
+
+
+class SignupFlow(Base):
+    """
+    Signup flows/incomplete users
+
+    Coinciding fields have the same meaning as in User
+    """
+
+    __tablename__ = "signup_flows"
+
+    id = Column(BigInteger, primary_key=True)
+
+    # housekeeping
     created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
-    expiry = Column(DateTime(timezone=True), nullable=False)
+    flow_token = Column(String, nullable=False, unique=True)
+    email_verified = Column(Boolean, nullable=False, default=False)
+    email_sent = Column(Boolean, nullable=False, default=False)
+    email_token = Column(String, nullable=True)
```

Comment:
nullable=False as we always sent out an email the first thing that happens.

Reply:
I got consistency errors when I set `nullable=False` and it is a bit of spagetti code to follow. Will try to restructure.

Reply:
Did you figure it out?

Reply:
Not yet, I think it has to do with refresh() setting it to null temporarily before the transaction is committed.

Reply:
I am travelling abroad and will not have time to work more. Looks good to merge and do cleanups later!

Reply:
OK, thanks. I'll take it from here!